code/game/objects/structures/window_frames.dm

```diff
@@ -185,8 +196,22 @@
 
 /obj/structure/window_frame/update_overlays()
 	. = ..()
-	if(has_grille)
-		. += mutable_appearance(grille_icon, "[grille_icon_state]-[smoothing_junction]")
+	if(!has_grille)
+		return
+
+	var/offset = GET_TURF_PLANE_OFFSET(src)
+	grill_overlay = mutable_appearance(grille_icon, "[grille_icon_state]-[smoothing_junction]", plane = GAME_PLANE, offset_const = offset)
+	grill_black_overlay = mutable_appearance(grille_black_icon, "[grille_icon_state]_black-[smoothing_junction]", plane = src.plane, offset_const = offset)
+	. += grill_overlay
+	. += grill_black_overlay
+
+	if(!frame_icon)
+		return
```

Comment:
could we have a frame icon but no grill?

Reply:
yes, ill make a proc for each